### ISSUE TITLE
Better debugging on segfault?

### ISSUE THREAD

**Issue by luc-tielen** [Primordus/lua-quickcheck#11]
Currently it's possible to test C code by using the Lua FFI.
C code can however trigger a segfault when a bug is triggered in a program.
It would be useful to still be able to capture this output even after the segfault occurs, right now only the following shows up (in Linux):

zsh: segmentation fault (core dumped)  LD_LIBRARY_PATH=spec/fixtures/ busted --helper=spec/fixtures/helper.lua 

Possible options to improve this:
- separate all C properties into a separate process, then monitor that process somehow for segfaults
- let it coredump, investigate core file afterwards

**Comment by ligurio** [comment]
> "separate all C properties into a separate process, then monitor that process somehow for segfaults"

this way looks preferable :)

**Comment by luc-tielen** [comment]
Yeah, looking back at it this is the way to go. But I'm currently not using the library so I probably won't be adding it any time soon 😅

**Comment by ligurio** [comment]
It's worth to look how it is implemented in theft (PBT library for C) - https://github.com/silentbicycle/theft/blob/master/doc/forking.md
theft uses timeouts and recommend to disabled crash reporter because it can slowdown testing process.
